### PR TITLE
Improve Javadoc on CucumberContextConfiguration

### LINKED ISSUE #2715: Improve Javadoc on CucumberContextConfiguration
### 👓 What did you see?

We use the `@CucumberContextConfiguration` to mark a class as the context configuration. Unfortunately the annotation doesn't explain this.

### 📚 Any additional context?

https://stackoverflow.com/questions/75858132/stumped-on-cucumberbackendexception-please-annotate-a-glue-class-with-some-cont/75858860#75858860

### PR DESCRIPTION
**Pull request by vishesh10**
### 🤔 What's changed?

Improved the Javadoc for CucumberContextConfiguration 

### ⚡️ What's your motivation? 

Fixes #2715

### 🏷️ What kind of change is this?

- :bank: Refactoring (improvement to documentation)

### 📋 Checklist:

- [x] I agree to respect and uphold the [Cucumber Community Code of Conduct](https://cucumber.io/conduct/)
- [x] My change requires a change to the documentation.
  - [x] I have updated the documentation accordingly.

----

### CODE REVIEW

**Comment by mpkorstanje** [comment]
Hey thanks for picking this up. Unfortunately it's not as easy as the last one. 

The Stackoverflow question I linked from in #2715 got was deleted. Nevertheless, in it a question was asked about the purpose and usage of the `@CucumberContextConfiguration`. This question should have been answered by the Javadoc.

The questions that should be answered are:

1. What does this annotation do.
2. How should it be used.
3. How does it work? 

You may find some guidance in the readme but it doesn't fully explain everything.

https://github.com/cucumber/cucumber-jvm/tree/main/cucumber-spring#configuring-the-test-application-context

**Comment by vishesh10** [comment]
@mpkorstanje,
Thanks for reviewing and providing the reference link. Will go through it and make the changes accordingly.

**Comment by vishesh10** [comment]
Hi @mpkorstanje,
I have added a commit with the new changes. 
Thanks

File: cucumber-spring/src/main/java/io/cucumber/spring/CucumberContextConfiguration.java [hunk 1]
```diff
@@ -8,7 +8,27 @@
...
 
 /**
- * Configure an application context class that Cucumber should use
+ * This annotation is used on a configuration class to make the Cucumber aware
+ * of the test configuration. This is to be used in conjunction with
+ * {@code @ContextConfiguration}, {@code @ContextHierarchy} or
+ * {@code @BootstrapWith}. In case of SpringBoot, the configuration class can be
+ * annotated as follows:
+ * <p>
+ * <pre>{@code
```
Comment (mpkorstanje) [inline cucumber-spring/src/main/java/io/cucumber/spring/CucumberContextConfiguration.java:17]: FYI: The magic syntax for multi-line code blocks is `<pre>{@code <the actual code>}</pre>.

**Review by mpkorstanje — COMMENTED** [review]
Cheers. I've merged the changes with some editing.

You initially put in most of the README in there. But that was a bit too much. The java doc should explain the purpose of the class not the entire module.

**Comment by vishesh10** [comment]
Thanks @mpkorstanje for reviewing the PR and providing the valuable inputs.

**Pull request merged**